dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+            suffix=$(echo "${suffix}" | head -1)
+        fi
+
+        if [[ "${suffix}" = *"${CONFIG_SELECTION_SUFFIX}"* ]]; then
+            var="${var}, exec"
+        fi
+        local result="${BASH_REMATCH[1]}${suffix}${BASH_REMATCH[2]}(${var})${BASH_REMATCH[4]}"
+        result=$(echo "${result}" | sed -E 's/\) *\(+/,/g;s/, *\)/\)/g')
+        if [[ "${suffix}" = "" ]]; then
+            # if the function does not exist in the file, comment it.
+            echo "//remove//${result}"
+        else
+            echo "${result}"
+        fi
+    fi
+}
+
+
+convert_regex_allowed() {
+    local str="$1"
+    str=$(echo "${str}" | sed -E 's/(\/|\(|\)|\.)/\\\1/g')
+    echo "$str"
+}
+
+# Transfer header file to the correct one
+# common -> dpcpp_common
+# cuda -> dpcpp
+# add sycl header
+
+input="$1"
+filename="${ROOT_DIR}/$input"
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "Porting file ${filename}"
+fi
+# check file exists
+
+if [ ! -f "${filename}" ]; then
+    echo "${filename} does not exist"
+    exit 1
+fi
+
+
+temp=""
+IN_SYNTAX="false"
+DEVICE_FILE=""
+GINKGO_LICENSE_BEACON="******************************<GINKGO LICENSE>******************************"
+UNFORMAT_FILE="unformat.cpp"
+FORMAT_FILE="format.cpp"
+cp "${filename}" "${UNFORMAT_FILE}"
+OUTPUT_FILE="source.cpp"
+EMBED_FILE="embed.cu"
+EMBED_HOST_FILE="embed_host.cu"
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "the original file ${UNFORMAT_FILE}"
+    echo "the formatted file ${FORMAT_FILE}"
+    echo "collect common/*.inc in file ${EMBED_FILE}"
+    echo "add autohost func in file ${EMBED_HOST_FILE}"
+    echo "convert original CUDA call in file ${OUTPUT_FILE}, which is the file for dpct"
+fi
+rm "${OUTPUT_FILE}"
+# echo "#include \"trick/dim3.hpp\"" >> "${OUTPUT_FILE}"
+echo "#define GET_QUEUE 0" >> "${OUTPUT_FILE}"
+# add empty ginkgo license such that format_header recognize some header before header def macro
+echo "/*${GINKGO_LICENSE_BEACON}" >> "${OUTPUT_FILE}"
+echo "${GINKGO_LICENSE_BEACON}*/" >> "${OUTPUT_FILE}"
+rm "${GLOBAL_FILE}"
+rm "${EMBED_FILE}"
+while IFS='' read -r line; do
+    if [[ "${line}" =~ $DEVICE_CODE_SYNTAX ]]; then
+        # hold the command to easy replace
+        echo "// ${line}" >> ${EMBED_FILE}
+        device_file="${BASH_REMATCH[1]}"
+        [ "${EXTRACT_KERNEL}" == "true" ] && echo "/**** ${device_file} - start ****/" >> "${EMBED_FILE}"
+        cat "${ROOT_DIR}/${device_file}" >> "${EMBED_FILE}"
+        [ "${EXTRACT_KERNEL}" == "true" ] && echo "/**** ${device_file} - end ****/" >> "${EMBED_FILE}"
+        if [ -n "${DEVICE_FILE}" ]; then
+            DEVICE_FILE="${DEVICE_FILE};"
+        fi
+        DEVICE_FILE="${DEVICE_FILE}${device_file}"
+    else
+        echo "${line}" >> "${EMBED_FILE}"
+    fi
+done < "${UNFORMAT_FILE}"
+${CLANG_FORMAT} -style=file "${EMBED_FILE}" > "${FORMAT_FILE}"
+${SCRIPT_DIR}/add_host_function.sh "${FORMAT_FILE}" > "${EMBED_HOST_FILE}"
+
```

Comment:
Maybe a newline and a comment here (correct it if I'm wrong)?
```# Add an extra host function so that the converted DPC++ code will look like CUDA.```